### PR TITLE
feat: add gorm plugin

### PR DESCRIPTION
**Pull request by kehuili**
Resolves [#36 ](https://github.com/SkyAPM/go2sky-plugins/issues/36)

### CODE REVIEW

File: gorm/gorm.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+		defer span.End()
+
+		sql := db.Statement.SQL.String()
+		vars := db.Statement.Vars
+		err := db.Statement.Error
+
+		span.SetComponent(s.getComponent())
+		span.SetSpanLayer(agentv3.SpanLayer_Database)
+		span.Tag(go2sky.TagDBType, string(s.sqlType))
```
Comment (arugal) [inline gorm/gorm.go:123]: I think the tag key should look [like this](https://github.com/SkyAPM/go2sky-plugins/blob/6562b6c6f82f96f019512f4a2414a828f63a8232/sql/common.go?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L37-L40), and please help update [go2sky](https://github.com/SkyAPM/go2sky/blob/master/span.go#L154) :)

File: gorm/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Go2Sky with Gorm
```
Comment (wu-sheng) [inline gorm/README.md:1]: I think this doc should be linked from https://github.com/SkyAPM/go2sky-plugins#trace-plugins

File: gorm/gorm.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+		span.Tag(go2sky.TagDBStatement, sql)
+
+		if len(vars) != 0 {
```
Comment (arugal) [inline gorm/gorm.go:126]: Add `reportParam` and `reportQuery` options [like this](https://github.com/SkyAPM/go2sky-plugins/blob/6562b6c6f8/sql/options.go#L34-L35), let the user choose. WDYT?

File: gorm/gorm.go (hunk 1 above)
Comment (kehuili) [reply gorm/gorm.go:123]: Do you mean to change this tag `TagDBBindVariables` in go2sky into `tagDbSqlParameters = "db.sql.parameters"`?

Comment (arugal) [reply gorm/gorm.go:123]: yes

**Comment by wu-sheng** [comment]
Should this be closed?

**Comment by kehuili** [comment]
> Should this be closed?

Can you reopen it, or should I raise another PR?

**Comment by wu-sheng** [comment]
I opened this.

**Review by arugal — APPROVED** [review]
LGTM

**Review by wu-sheng — APPROVED** [review]

**Comment by wu-sheng** [comment]
@arugal Let's release 1.3.0

**Pull request merged**